Feature/additional program associations

jayckaiser opened the pull request:
This PR contains base and stage models for three programAssociation resources:

- studentLanguageInstructionProgramAssociations
- studentHomelessProgramAssociations
- studentTitleIPartAProgramAssociations

I modeled the structure of each off of the existing `studentProgramAssociations` and `studentSpecialEducationProgramAssociations`, preferring naming conventions in `studentProgramAssociations`. I also created a couple of staging models for unnesting several of the list columns in base.

I'm opening this PR to begin a discussion about the structure of these models and to elicit feedback. I will begin work on the `edu_wh` models downstream once these have been reviewed.

ejoranlienea reviewed (changes requested):
Looking really good. Two small tweaks:
1. Can you add lines to the Changelog? Will need to add a new top level header for 'Unreleased Changes' and the three subheaders, with some bullets under new features for this stuff.
2. In the dedupes, `delivering_ed_org_id` is redundant with `k_program`, and `school_year` is redundant with `k_student`, so I think those can go.

Also, there are some minor inconsistencies with special education:
- `delivering_ed_org_id` vs `ed_org_id`
- `spec_ed_program_begin_date` vs `program_enroll_begin_date`
These should probably be brought into alignment. Having consistent begin/end date names across program types makes sense, but we should check for downstream usage of this field in edu_wh before making changes.

jayckaiser commented:
I have removed `delivering_ed_org_id` from the deduplicate CTE of all programAssociations staging models, and I have updated the changelog with these updates.

Addressing the inconsistencies between these `programAssociations` models and `studentSpecialEducationProgramAssociations` should be completed in a separate branch, as this will require updating `edu_wh` as well.
